src/control/deleteCollection.ts

```diff
@@ -0,0 +1,46 @@
+import { IndexOperationsApi } from '../pinecone-generated-ts-fetch';
+import type { ResponseError } from '../pinecone-generated-ts-fetch';
+import { mapHttpStatusError } from '../errors';
+import { builOptionConfigValidator } from '../validator';
+import { Static, Type } from '@sinclair/typebox';
+import { validCollectionMessage } from './utils';
+
+const DeleteCollectionIndex = Type.String({ minLength: 1 });
+export type CollectionName = Static<typeof DeleteCollectionIndex>;
+
+export const deleteCollection = (api: IndexOperationsApi) => {
+  const validator = builOptionConfigValidator(
+    DeleteCollectionIndex,
+    'deleteCollection'
+  );
+
+  return async (collectionName: CollectionName): Promise<void> => {
+    validator(collectionName);
+
+    try {
+      await api.deleteCollection({ collectionName: collectionName });
+      return;
```

Comment:
The overall return value here is `Promise<void>` which I think makes sense. I considered this question when I was implementing and couldn't think of anything very meaningful to return; on success the underlying generated openapi client just returns `''` when deleting which I don't think adds any information and can actually be misleading because `''` is a falsy value even though it succeeded.

If the request fails, this will throw and the promise will be rejected. Some APIs echo back the name/id of the thing that was deleted or a delete count when deleting multiple records, but our API doesn't return those today and to me it doesn't seem that useful anyway. As a caller, if I ask an api to return collection `'foo'` and it says it succeeded, I already know everything I need to know about the state of the world.